start.sh

```
@@ -427,28 +435,88 @@ EOF
+# restart policy (README examples use --restart unless-stopped) revives it.
+# ----------------------------------------------------------------------------
+set +e
+
+_shutdown() {
+    trap - TERM INT
+    log "INFO: Stop signal received, shutting down HaRP processes..."
+    [ -n "${WATCHDOG_PID:-}" ] && kill "$WATCHDOG_PID" 2>/dev/null
+    kill -TERM "$HAPROXY_PID" "$AGENT_PID" "$FRPS_PID" ${FRPC_PID:+"$FRPC_PID"} 2>/dev/null
+    wait
+    exit 0
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)